### PR TITLE
Update transform.py

### PR DESCRIPTION
**Pull request by pjleiwa6g**
For Snowflake query_tag set to value which is entered in the jdbc params of the snowflake destination connector. If no value is set, the default value 'normalization' will be used.

## What
In the Snowflake Destination Connector jdbc parameters can be set. One of the parameters can be the query_tag. This works fine when using the raw option. But with basic normalization the query_tag will be set to 'normalization' in transfrom.py
Now, first a check done to see if a value is set for query_tag, if so this value will be used. In the other case the default value 'normalization' will be used.

## How
A new function has been added, get_jdbc_param_value, which can lookup a paramater value of a given parameter. In this case it is the parameter 'query_tag'.
If this parameter has a value, this value will be returned if not, 'normalization' will be returned.

## Recommended reading order
1. `transformation.py

## 🚨 User Impact 🚨
If the user does not change the settings in the Snowflake Destination Connector, things will be as is -> in case of normalization the default value will be used. Only if the user sets a specfic value for query_tag, that value will be used.

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks @pjleiwa6g I'll review your contribution tomorrow. 

 From Zendesk 6171528851725

File: airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py [hunk 1]
```diff
@@ -224,6 +226,17 @@ def transform_snowflake(config: Dict[str, Any]):
...
+    def get_jdbc_param_value(parameter,jdbc_url_params,default_value):
+        parameter_length=len(parameter) + 1 # +1 because of the = sign
+        parameters=jdbc_url_params.split(',')
+        parameter_value=default_value
+        
+        for i in parameters:
+            if i.startswith(parameter):
+                parameter_value=urllib.parse.unquote(i[parameter_length:])
+                                
+                return parameter_value
```
Comment (marcosmarxm) [inline airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py:239]: This is going to return only the first parameter.
Also if don't find any parameter won't return the default value.

Please add unit test for this function.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Reviewed @pjleiwa6g sorry the delay. Need some changes.

File: airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py [hunk 2]
```diff
@@ -194,6 +195,7 @@ def transform_snowflake(config: Dict[str, Any]):
         print("transform_snowflake")
         # here account is everything before ".snowflakecomputing.com" as it can include account, region & cloud environment information)
         account = config["host"].replace(".snowflakecomputing.com", "").replace("http://", "").replace("https://", "")
+        query_tag=TransformConfig.get_jdbc_param_value("query_tag".upper(),config["jdbc_url_params"].upper(),"normalization")
```
Comment (marcosmarxm) [inline airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py:198]: Please add snowflake dbt url docs here.

File: airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py (hunk 1 above)
Comment (pjleiwa6g) [reply airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py:239]: Thanks, I will take a look at it and adjust it.

**Comment by marcosmarxm** [comment]
Also @pjleiwa6g would be good to update the spec.json file to make clear to users that the `jdbc_url_param` will be used in normalization process with dbt.

**Comment by marcosmarxm** [comment]
Closing due inactivity. Please let me know if you want to continue the work @pjleiwa6g

**Pull request closed without merging**